sandbox/apparmor/notify/listener/listener.go

```diff
@@ -431,6 +542,38 @@ func (l *Listener) newRequest(msg notify.MsgNotificationGeneric) (*Request, erro
 	}, nil
 }
 
+// decrementPendingCheckFinal decrements the pending count if it's not already
+// 0, and returns whether this was the final pending request.
+//
+// The caller should only call this method if the message associated with the
+// request *was* marked by the kernel as having been previously sent.
+func (l *Listener) decrementPendingCheckFinal() (isFinal bool) {
+	l.pendingMu.Lock()
```

Comment:
This function could be lockless: https://pkg.go.dev/sync/atomic@go1.24.3#Int32 (since go 1.19) or https://pkg.go.dev/sync/atomic@go1.24.3#AddInt32 family